catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/cql/MessageValidatorManagerTest.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.cql;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.validation.MessageValidator;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Test;
+
+public class MessageValidatorManagerTest {
+
+  private final ProcessingDetails mockedProcessingDetailsWithNullSourceId =
+      mock(ProcessingDetails.class);
+
+  private final ProcessingDetails mockedProcessingDetailsFromTestSource =
+      mock(ProcessingDetails.class);
+
+  private final MessageValidator mockedValidatorWithNullId = mock(MessageValidator.class);
+
+  private final MessageValidator mockedValidatorOfNotTestSource = mock(MessageValidator.class);
+
+  private final MessageValidator firstMockedValidatorOfTestSource = mock(MessageValidator.class);
+
+  private Set validWarningsFromRunOfThisTest;
```

Comment:
We shouldn't use a raw `Set`.

Reply:
Why shouldn't we? I didn't think I should couple the implementation of `Set` which `MessageValidatorManager` returns to the type of `validWarningsFromRunOfThisTest`. Not that such a coupling would create much of a problem, but I don't know how the use of a raw `Set` here causes a problem.